### ISSUE TITLE
Add new vocabs

### ISSUE THREAD

**Issue by nuwanq** [cl-tohoku/bert-japanese#4]
What is the easiest way to add specific domain content to bert-japanese?
Thanks in advance.

**Comment by singletongue** [comment]
I'd like to know about your situation more specifically.
For example, what kind of texts are you trying to handle?
How are the sentences tokenized with the pretrained models' tokenizers?

**Comment by nuwanq** [comment]
> I'd like to know about your situation more specifically.
> For example, what kind of texts are you trying to handle?
> How are the sentences tokenized with the pretrained models' tokenizers?

Thank you for the reply.
I am dealing with some email corpus in Japanese. There are some specific / special words in the corpus.
For example lets' assume that I need to predict MASK words or predict next word.
I am using Mecab with neologd.

In that case I'm going to retrain my corpus.  (Can I just fine tune in this case without pretrain again?)

**Comment by singletongue** [comment]
Thank you for sharing your needs.

It should be possible to expand vocabulary and the corresponding embedding layer.
But it can be very tricky to implement that functionality.

By the way, have you tried any tokenization with the pretrained vocabulary?
For example (using Transformers by Hugging face):
```py
from transformers.tokenization_bert_japanese import BertJapaneseTokenizer
tokenizer = BertJapaneseTokenizer.from_pretrained('bert-base-japanese')
tokenizer.tokenize('今日はとーってもいい日でした(^^)')
```

If you tokenize a text you want to handle and get many [UNK]s, then the difference of vocabulary matters.
However, I don't really think it's common because the tokenization algorithm in BERT is sub-word based.

**Comment by tomohideshibata** [comment]
I'm not sure I can understand what you want to do, but we can use an `add_tokens` function in a tokenizer when we want to add some words to a vocabulary for fine-tuning. Please see the following manual.
https://huggingface.co/transformers/main_classes/tokenizer.html#transformers.PreTrainedTokenizer.add_tokens

**Comment by nuwanq** [comment]
> I'm not sure I can understand what you want to do, but we can use an `add_tokens` function in a tokenizer when we want to add some words to a vocabulary for fine-tuning. Please see the following manual.
> https://huggingface.co/transformers/main_classes/tokenizer.html#transformers.PreTrainedTokenizer.add_tokens

Thank you

**Comment by nuwanq** [comment]
> Thank you for sharing your needs.
> 
> It should be possible to expand vocabulary and the corresponding embedding layer.
> But it can be very tricky to implement that functionality.
> 
> By the way, have you tried any tokenization with the pretrained vocabulary?
> For example (using Transformers by Hugging face):
> 
> ```python
> from transformers.tokenization_bert_japanese import BertJapaneseTokenizer
> tokenizer = BertJapaneseTokenizer.from_pretrained('bert-base-japanese')
> tokenizer.tokenize('今日はとーってもいい日でした(^^)')
> ```
> 
> If you tokenize a text you want to handle and get many [UNK]s, then the difference of vocabulary matters.
> However, I don't really think it's common because the tokenization algorithm in BERT is sub-word based.

Thank you for sharing your knowledge. I ended up training with own domain corpus with the help of google TPU.
